src/frontend/AppContainerWrapper.tsx

```diff
@@ -99,6 +105,26 @@ const AppContainerWrapper = () => {
   const [{ experiments }] = React.useContext(SettingsContext);
   const [hidePracticeBar, setHidePracticeBar] = React.useState(true);
   const [hidePracticeMode, setHidePracticeMode] = React.useState(false);
+  const [{ authStatus }, setAuthState] = React.useContext(SecurityContext);
+
+  React.useEffect(() => {
+    const appStateListener = AppState.addEventListener(
+      "change",
+      (nextAppState: AppStateStatus) => {
+        if (authStatus !== "notRequired") {
+          if (nextAppState === "inactive" || nextAppState === "background") {
+            setAuthState({ type: "setAuthStatus", newAuthStatus: "pending" });
+          }
+        }
+      }
+    );
+
+    return () => appStateListener.remove();
+  }, [authStatus]);
```

Comment:
sorry i forgot to add this in my previous suggestion but i _think_ you need to add `setAuthState` here too?

```suggestion
  }, [authStatus. setAuthState]);
```